src/DependencyInjection/DoctrineExtension.php

```diff
@@ -429,6 +429,21 @@ private function detectMappingType(string $directory, ContainerBuilder $containe
         return $type;
     }
 
+    /**
+     * Check if the file content contains a class-like annotation
+     *
+     * @internal
+     */
+    public static function textContainsAnnotation(string $quotedMappingObjectName, string $content): bool
+    {
+        return preg_match('/^(?:[ ]\*|\/\*\*)[ ]@                # Match phpdoc start or line with an at
+            \\\\?                                               # Can start with antislash
+            ([a-zA-Z_\x80-\xff][a-zA-Z0-9_\x80-\xff]*\\\\?)*    # Match 0-n namespace, the antislash is optionnal as it may be a prefix if an alias is used
```

Comment:
-1 for the part about the prefix thing. This leads to matching *anyway* annotation that has `Entity` somewhere in its name).

We should keep checking for the `Entity` word